### PR TITLE
[FOR-1841] Add support for userId param

### PR DESCRIPTION
**Pull request by djoksimo**
# Description

- Add required `userId` to `tokenizeEBTCard`

## Also
- [x] Add tests related to the `userId` changes
- [ ] ~~Add KDoc style comments to clarify userId~~ Deferring to a later ticket/PR

## Notes
- Describe your changes

## Tests Added / Updated?
- YES 

- Pass `userId` in tests
- ~~Manually tested by completing full checkout where `userId` is provided vs. not provided~~
- ~~Verified that as a consumer of the Android SDK, `userId` is truly optional~~

## Rollout

⚠️  do not release this PR until #33 and then #32 are merged down too

### CODE REVIEW

**Review by owenkim — COMMENTED** [review]
Change looks good but I'm curious what the integration test failure is.

File: forage-android/src/main/java/com/joinforage/forage/android/network/TokenizeCardService.kt [hunk 1]
```diff
@@ -16,23 +16,24 @@ internal class TokenizeCardService(
...
         val requestBody =
-            PaymentMethodRequestBody(cardNumber = cardNumber).toJSONObject().toString()
+            PaymentMethodRequestBody(cardNumber = cardNumber, userId = userId).toJSONObject().toString()
 
         val body: RequestBody =
             requestBody.toRequestBody("application/json; charset=utf-8".toMediaTypeOrNull())
 
         val request: Request = Request.Builder()
             .url(url)
+            .header(ForageConstants.Headers.API_VERSION, "2023-03-31")
```
Comment (ForageRob) [inline forage-android/src/main/java/com/joinforage/forage/android/network/TokenizeCardService.kt:36]: Version header may do better in the global part of the http client, see here: https://github.com/teamforage/forage-android-sdk/blob/31ea5a82cf20bc0e6b3c65f2263f43f8951e55c4/forage-android/src/main/java/com/joinforage/forage/android/network/OkHttpClientBuilder.kt#L17

**Review by ForageRob — CHANGES_REQUESTED** [review]
Minor comment on placement of version header but overall LGTM

File: forage-android/src/main/java/com/joinforage/forage/android/network/TokenizeCardService.kt (hunk 1 above)
Comment (djoksimo) [reply forage-android/src/main/java/com/joinforage/forage/android/network/TokenizeCardService.kt]: Addressed here: a480072

Comment (djoksimo) [reply forage-android/src/main/java/com/joinforage/forage/android/network/TokenizeCardService.kt]: Should this request stay at version: "2023-02-01"?

https://github.com/teamforage/forage-android-sdk/blob/31ea5a82cf20bc0e6b3c65f2263f43f8951e55c4/forage-android/src/main/java/com/joinforage/forage/android/network/MessageStatusService.kt#L25

Comment (dleis612) [reply forage-android/src/main/java/com/joinforage/forage/android/network/TokenizeCardService.kt]: @ForageRob unless we determined we are changing how our versioning currently works, we can't pass the header globally. All endpoints without a version `2023-03-31` will fail. Did we decide to update this?

@djoksimo that endpoint still needs the version header. It is an internal endpoint that I versioned a while ago to change the response body structure.

**Review by dleis612 — CHANGES_REQUESTED** [review]

File: forage-android/src/main/java/com/joinforage/forage/android/network/TokenizeCardService.kt (hunk 1 above)
Comment (djoksimo) [reply forage-android/src/main/java/com/joinforage/forage/android/network/TokenizeCardService.kt]: Reverted/dropped the commit so now only the `POST payment_methods` has this API_VERSION attached

File: forage-android/src/main/java/com/joinforage/forage/android/network/OkHttpClientBuilder.kt [hunk 2]
```diff
@@ -14,6 +14,10 @@ internal object OkHttpClientBuilder {
                 Interceptor { chain ->
                     val request = chain.request()
                         .newBuilder()
+                        .addHeader(
+                            ForageConstants.Headers.API_VERSION,
```
[the remark this answers is not in the document]
Comment (djoksimo) [reply forage-android/src/main/java/com/joinforage/forage/android/network/OkHttpClientBuilder.kt:18]: https://github.com/teamforage/forage-android-sdk/pull/27#discussion_r1165957516

**Review by dleis612 — APPROVED** [review]
This gets a contingent +1 on finally getting the qa tests to pass, which I can help with tomorrow.

**Comment by dleis612** [comment]
> This gets a contingent +1 on finally getting the qa tests to pass, which I can help with tomorrow.

The current issue is that staging doesn't have the backend changes required to pass the qa-tests. Once that is deployed, we can test here.

**Pull request merged**